estimate_infections_weekly_data.R

```diff
@@ -0,0 +1,237 @@
+withr::local_options(list(mc.cores = parallel::detectCores() - 1))
+
+# Use the example data for confirmed cases from EpiNow2
+reported_cases <- EpiNow2::example_confirmed
+reported_cases_weekly <- data.table::copy(reported_cases)
+
+# Aggregate the daily cases to weekly cases (sum of daily cases)
+reported_cases_weekly[, confirm := frollsum(confirm, 7)]
+reported_cases_weekly <- reported_cases_weekly[seq(7, nrow(reported_cases_weekly), 7)]
+
+# ============================================================================== #
+# DEFINE EPIDEMIOLOGICAL PARAMETERS AND DISTRIBUTIONS
+# ============================================================================== #
+
+# Extract distribution the incubation period for COVID-19
+covid_incubation_dist <- epiparameter::epiparameter_db(
+  disease = "covid",
+  epi_name = "incubation",
+  single_epiparameter = TRUE
+)
+
+# Extract the serial interval distribution
+serial_interval_dist <- epiparameter::epiparameter_db(
+  disease = "covid",
+  epi_name = "serial",
+  single_epiparameter = TRUE
+)
+
+# ============================================================================== #
+# ESTIMATE INFECTIONS AND Rt WITH EPINOW2
+# ============================================================================== #
+
+# Extract parameters and maximum of the distribution for EpiNow2
+incubation_params <- epiparameter::get_parameters(covid_incubation_dist)
+incubation_max_days <- round(quantile(covid_incubation_dist, 0.999)) # Upper 99.9% range needed for EpiNow2
+
+# Create a LogNormal object for the incubation period
+incubation_lognormal <- EpiNow2::LogNormal(
+  meanlog = incubation_params[["meanlog"]],
+  sdlog = incubation_params[["sdlog"]],
+  max = incubation_max_days
+)
+
+# Extract parameters and maximum of the distribution for EpiNow2
+serial_interval_params <- epiparameter::get_parameters(serial_interval_dist)
+serial_interval_max_days <- round(quantile(serial_interval_dist, 0.999)) # Upper 99.9% range needed for EpiNow2
+
+# Create a LogNormal object for the serial interval
+serial_interval_lognormal <- EpiNow2::LogNormal(
+  meanlog = serial_interval_params[["meanlog"]],
+  sdlog = serial_interval_params[["sdlog"]],
+  max = serial_interval_max_days
+)
+
+# Create data with missing dates filled in for EpiNow2
+input_data_epinow <- EpiNow2::fill_missing(
+  reported_cases_weekly,
+  missing_dates = "accumulate",
+  initial_accumulate = 1 # Don't model the first data point (to match EpiEstim method)
+)
+
+# Estimate infections using EpiNow2
+estimates_epinow <- EpiNow2::epinow(
+  data = input_data_epinow,
+  generation_time = generation_time_opts(serial_interval_lognormal),
+  forecast = forecast_opts(horizon = 0, accumulate = 1), # Forecasting is turned off to match with EpiEstim
+  rt = rt_opts(
+    prior = Gamma(mean = 5, sd = 5) # same prior as used in EpiEstim default
+  ),
+  CrIs = c(0.025, 0.05, 0.25, 0.75, 0.95, 0.975), # same prior as used in EpiEstim default
```

Comment:
I tested this in the cloud, and the running time dropped from ~600 seconds to ~200 seconds.
```suggestion
  CrIs = c(0.025, 0.05, 0.25, 0.75, 0.95, 0.975), # same prior as used in EpiEstim default
  stan = EpiNow2::stan_opts(samples = 1000, chains = 3), # reduce computation time to ~3 minutes
```

Reply:
Awesome, it can be a minimum of 2 so you could try that as well so we see the most optimal since this is for demonstrative purposes. We could leave a comment saying it should be reverted to 4 chains for better inference.

Reply:
```
> tictoc::toc()
113.193 sec elapsed
```
works well
```suggestion
  CrIs = c(0.025, 0.05, 0.25, 0.75, 0.95, 0.975), # same prior as used in EpiEstim default
  stan = EpiNow2::stan_opts(samples = 1000, chains = 2), # revert to 4 chains for better inference
```